src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -35,13 +35,46 @@ function useDecline () {
   })
 }
 
+function useFetchCounterparties () {
+  const { data: { holofuelActionableTransactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
+
+  const { loading, error, data: { holofuelInboxCounterparties } = {}, client } = useQuery(HolofuelInboxCounterpartiesQuery)
+
+  if (holofuelInboxCounterparties) {
+    const filterTransactionsByAgentId = (agent, txListType) => txListType.filter(transaction => transaction.counterparty.id === agent.id)
+    const updateTxListCounterparties = (txListType, counterpartyList) => counterpartyList.map(agent => {
+      const matchingTx = filterTransactionsByAgentId(agent, txListType)
+      return matchingTx.map(transaction => { Object.assign(transaction.counterparty, agent); return transaction })
+    })
+
+    const result = flatten(updateTxListCounterparties(holofuelActionableTransactions, holofuelInboxCounterparties))
+
+    client.writeQuery({
+      query: HolofuelActionableTransactionsQuery,
+      data: {
+        holofuelActionableTransactions: result
+      }
+    })
+  }
+
+  let response
+  if (loading) response = { loading: true }
+  else if (error) response = { error: `Error: ${error}` }
+  else response = holofuelInboxCounterparties
+  return response
+}
```

Comment:
given that we don't use this return value, let's omit it, and maybe put a comment in its place `// this function updates the cache of other Queries, we don't use a return value directly`